scripts/serve/serve-static.js

```diff
@@ -55,36 +65,54 @@ const virtualApp = (site, index, fs) => {
     // Setup a default index for the server.
     if (filename === '/') {
       filename = `/${index}`;
+    } else if (filename === '/favicon.ico') {
+      res.sendStatus(200);
+      return;
     }
 
     const filepath = `${site}${filename}`;
 
     if (fs.existsSync(filepath)) {
       res.setHeader('content-type', mime.contentType(path.extname(filename)));
-      res.send(fs.readFileSync(filepath));
-    } else if (filename === '/favicon.ico') {
-      res.sendStatus(200);
+
+      let fileContent = fs.readFileSync(filepath, 'utf8');
+
+      const langPattern = /lang="[a-zA-Z0-9-]*"/;
+      const langLocale = `lang="${locale}"`;
+
+      const isLanguageSet = fileContent.match(langPattern);
+
+      // Set the test locale for the file
+      if (isLanguageSet) {
+        fileContent = fileContent.replace(langPattern, langLocale);
+      } else {
+        fileContent = fileContent.replace(/<html/, `<html ${langLocale}`);
```

Comment:
This might be dangerously assuming that only html pages will be served, which isn't always the case. We're serving mock avatar assets with serve-static, for example.

Reply:
Nope, the only thing we're serving (right now at least) is images. Just wanted to make sure this wouldn't break anything, it might be fine as-is.

Reply:
Lets only do this replace if the file extension is .html. This may get confusing to consumers since normal web servers don't manipulate the files they're serving. At a minimum lets also add a comment to the file we're manipulating explaining where this file came from. Also break this logic out into it's own function this function is getting to long.